onnx/defs/tensor/defs.cc

```diff
@@ -492,7 +492,33 @@ Takes a  parameter `axes` with a list of axes to squeeze.
     .TypeConstraint(
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain input and output types to float tensors.");
+        "Constrain input and output types to float tensors.")
+    .ShapeInferenceFunction([](InferenceContext& ctx) {
+        if (!hasExactlyNInputTypes(ctx, 1, "Squeeze")) {
+          return;
+        }
+
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+        std::vector<int64_t> axes;
+        if (!getRepeatedAttribute(ctx, "axes", axes)) {
```

Comment:
`axes` is required attribute

Reply:
so - would you like me to throw an exception, or just assume it succeeds and remove the check? or leave it as is?

Reply:
we should have assertion here, but I know assertion will cause Appveyor hanging...

Reply:
throw an exception should be fine, because model is incorrect.

Reply:
yes, please throw an exception, or use this code: https://github.com/onnx/onnx/blob/master/onnx/common/assertions.h

ONNX_ASSERT is actually an exception, not a real assert

Reply:
IMHO, just returning is fine too. I think checking the model for correctness is not the job of shape-inference. It is intended for optimization, and it is fine if doesn't infer the output shape under some conditions.

Reply:
exactly because "checking the model for correctness is not the job of shape-inference", here should throw instead of silently return.